api/app/Plousio/Utils/StripeUtils.php

```
@@ -5,25 +5,58 @@
+    // credit & debit card
+    const STRIPE_CARD_FEE_BASE = 30;
+    const STRIPE_CARD_FEE_RATE = 0.029;
+    // ACH
+    const STRIPE_BANK_FEE_MAX = 500;
+    const STRIPE_BANK_FEE_RATE = 0.008;
 
     public static function dollarsToCents(string $amount): string
     {
         $clean = preg_replace("/[^0-9.]/i", '', $amount);
         $realNumber = (float)$clean;
-        return (string)($realNumber * 100);
+        return (int)($realNumber * 100);
     }
 
-    public static function validateCentsAmount(string $amountInCents, Offer $offer): bool
+    public static function validateCentsAmount(int $amountInCents, Offer $offer): bool
     {
         $maximumPlousioCommission = $offer->loanAmount * 0.12;
         $maximumPartnerCommission = $maximumPlousioCommission * 0.2;
 
         return ($amountInCents / 100) <= $maximumPartnerCommission;
     }
 
-    public static function calculateApplicationFee(string $amountInCents)
+    public static function calculateApplicationFee(int $amountInCents, string $type): int
     {
-        return $amountInCents * self::APPLICATION_FEE;
+        $fee = ceil($amountInCents * self::PLOUSIO_APPLICATION_FEE);
+
+        $fee = $fee + self::calculateStripeRegularFee($amountInCents, $type);
+
+        $fee = $fee + self::calculateStripeConnectFee($amountInCents, $fee);
+
+        return min($fee, $amountInCents);
+    }
+
+    public static function calculateStripeRegularFee(int $amountInCents, string $type): int
+    {
+        switch ($type) {
+            case 'card':
+                return ceil(($amountInCents * self::STRIPE_CARD_FEE_RATE) + self::STRIPE_CARD_FEE_BASE);
+            case 'bitcoin':
+                return ceil(min($amountInCents * self::STRIPE_BANK_FEE_RATE, self::STRIPE_BANK_FEE_MAX));
+            case 'bank_account':
+                return ceil(min($amountInCents * self::STRIPE_BANK_FEE_RATE, self::STRIPE_BANK_FEE_MAX));
+            default:
+                throw new \Exception('Type not supported');
```

You should throw an `\InvalidArgumentException` . It might be useful to print the attempted type in the message too.

This case should also be tested. you can test exceptions in phpunit with https://phpunit.de/manual/current/en/writing-tests-for-phpunit.html#writing-tests-for-phpunit.exceptions.examples.ExceptionTest2.php